mlflow/tensorflow/__init__.py

```diff
@@ -888,6 +996,99 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
         if metric_key is not None:
             metrics_logger.record_metrics(restored_metrics, stopped_epoch + 1)
 
+    def _log_keras_model(history, args):
+        def _get_input_data_slice():
+            """
+            Generates a sample ndarray or dict (str -> ndarray)
+            from the input type 'x' for keras ``fit`` or ``fit_generator``
+
+            :return: a slice of type ndarray or
+                     dict (str -> ndarray) limited to ``INPUT_EXAMPLE_SAMPLE_ROWS``
+            """
+            input_training_data = args[0]
+            input_example_slice = None
+            if isinstance(input_training_data, tensorflow.keras.utils.Sequence):
+                input_training_data = input_training_data[:][0]
+
+            if isinstance(input_training_data, np.ndarray):
+                input_example_slice = input_training_data[:INPUT_EXAMPLE_SAMPLE_ROWS]
+            elif isinstance(input_training_data, tensorflow.data.Dataset):
+                steps = 1
+                if history.params is not None and "steps" in history.params:
+                    steps = history.params["steps"]
+
+                def _extract_n_steps(input_example_n_steps):
+                    # Example:
+                    # when the tensorflow.data.Dataset is batched
+                    # (Total number of steps (batches of samples)),
```

Comment:
Still having some trouble interpreting what this means. Can we give a concrete example in the inline comment?